core/domain/feedback_services_test.py

```diff
@@ -28,30 +28,37 @@
 from core.platform import models
 from core.tests import test_utils
 
-(
-    feedback_models, email_models, suggestion_models
-) = models.Registry.import_models([
-    models.NAMES.feedback, models.NAMES.email, models.NAMES.suggestion
+from typing import List, Optional
+from typing_extensions import Final, TypedDict
+
+MYPY = False
+if MYPY:  # pragma: no cover
+    from mypy_imports import feedback_models
+    from mypy_imports import suggestion_models
+
+(feedback_models, suggestion_models) = models.Registry.import_models([
+    models.NAMES.feedback,
+    models.NAMES.suggestion
 ])
 
 
 class FeedbackServicesUnitTests(test_utils.EmailTestBase):
     """Test functions in feedback_services."""
 
-    USER_EMAIL = 'user@example.com'
-    USER_USERNAME = 'user'
-    EXP_1_ID = 'exp_1_id'
+    USER_EMAIL: Final = 'user@example.com'
+    USER_USERNAME: Final = 'user'
+    EXP_1_ID: Final = 'exp_1_id'
 
-    def setUp(self):
+    def setUp(self) -> None:
         super(FeedbackServicesUnitTests, self).setUp()
         self.signup(self.USER_EMAIL, self.USER_USERNAME)
-        self.user_id = self.get_user_id_from_email(self.USER_EMAIL)
+        self.user_id = self.get_user_id_from_email(self.USER_EMAIL)  # type: ignore[no-untyped-call]
 
-    def test_feedback_ids(self):
+    def test_feedback_ids(self) -> None:
         """Test various conventions for thread and message ids."""
         exp_id = '0'
         feedback_services.create_thread(
-            'exploration', exp_id, None, 'a subject', 'some text')
+            'exploration', exp_id, 'test_user', 'a subject', 'some text')
```

Comment:
It would be helpful if you guys reply to this ( @seanlip @vojtechjelinek @aasiffaizal )

Reply:
It's possible for someone to anonymously submit feedback through the Web UI, right? (You can see this by going to an incognito window, opening an exploration, and using the feedback tooltip in the top right next to the language selector.)

Reply:
Yeah, you are right!! 

So, Optional[str] is added for those methods that can accept `None` for anonymous users.

Also, I just kept `None` only for those test cases where we are are covering anonymous user test case, and for other test cases i just used `test_user` ( because previously in almost all test cases we are using none, which seems like we testing only for anonymous user. Please correct me if I'm wrong somewhere )

Reply:
What you've done is perfect -- thanks @sahiljoster32. Only thing I noticed is that you forgot to update the typeinfo in the docstrings of the incorrect methods?

Reply:
typeinfo is added in docstrings!!